graviti/dataframe/frame.py

```diff
@@ -339,18 +339,30 @@ def _extend(self, values: "DataFrame") -> None:
         for key, value in self._columns.items():
             value._extend(values[key])  # pylint: disable=protected-access
 
-    def _to_patch_data(self) -> List[Dict[str, Any]]:
-        patch_data = []
-        for record_key, values in zip(
-            self._record_key.to_pylist(),  # type: ignore[union-attr]
-            zip(*(column.to_pylist() for column in self._columns.values())),
-        ):
-            row_patch_data = dict(zip(self.schema, values))
-            for name in self._name:
-                row_patch_data = {name: row_patch_data}
-            row_patch_data[RECORD_KEY] = record_key
-            patch_data.append(row_patch_data)
-        return patch_data
+    def _to_patch_data(self, need_record_key: bool = True) -> List[Dict[str, Any]]:
+        if need_record_key:
+            patch_data = []
+            for record_key, values in zip(
+                # pylint: disable=protected-access
+                self._record_key._to_patch_data(),  # type: ignore[union-attr]
+                zip(*(column._to_patch_data() for column in self._columns.values())),
+            ):
+                row_patch_data = dict(zip(self.schema, values))
+                for name in self._name:
+                    row_patch_data = {name: row_patch_data}
+                row_patch_data[RECORD_KEY] = record_key
+                patch_data.append(row_patch_data)
+            return patch_data
+
+        return [
+            dict(zip(self.schema, values))
```

Comment:
get keys from schema is slow (for now), better to cache the keys

Reply:
refactoring by https://github.com/Graviti-AI/graviti-python-sdk/pull/435